README.md

```diff
@@ -1 +1,30 @@
-# indexer
\ No newline at end of file
+# indexer
+
+## Configuration
+You can configure the application in 3 ways.
+The order of priority is
+1. Command line arguments
+2. Environment variables
+3. Configuration files
+
+### Configuration using configuration files
+The default configuration file is `configs/config.yml`. You can use the `--config` flag to specify a different configuration file.  
+If you want to add secrets to the configuration file, you can copy `configs/secrets.example.yml` to `configs/secrets.yml` and add the secrets.
+
+### Configuration using environment variables
+You can also configure the application using environment variables. You can configure any configuration in the `config.yml` file using environment variables.  
+For example to configure the `rpc.url` configuration to `https://my-rpc.com`, you can set the `RPC_URL` environment variable to `https://my-rpc.com`.  
+
+### Configuration using command line arguments
+You can also configure the application using command line arguments.  
+For example to configure the `rpc.url` configuration to `https://my-rpc.com`, you can use the `--rpc-url` command line argument.
+Currently supported command line arguments are
+
+```
+--config
+--rpc-url
+--log-level
+--poller
+--commiter
+--failure-recoverer
```

Comment:
describe what each var accept and why it's needed

Reply:
updated readme